test/problems/second_quantization/lattice/models/test_heisenberg_model.py

```diff
@@ -0,0 +1,157 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test HeisenbergModel."""
+
+from test import QiskitNatureTestCase
+from retworkx import PyGraph, is_isomorphic
+from qiskit_nature.second_q.properties.lattices import Lattice
+from qiskit_nature.second_q.properties.lattices import LineLattice
+from qiskit_nature.second_q.properties import HeisenbergModel, IsingModel
+
+
+class TestHeisenbergModel(QiskitNatureTestCase):
+    """TestHeisenbergModel"""
+
+    def test_init(self):
+        """Test init."""
+        line = LineLattice(num_nodes=2)
+        heisenberg_model = HeisenbergModel(lattice=line)
+
+        with self.subTest("Check the graph."):
+            self.assertTrue(
+                is_isomorphic(
+                    heisenberg_model.lattice.graph, line.graph, edge_matcher=lambda x, y: x == y
+                )
+            )
+
+        with self.subTest("Check the second q op representation."):
+            terms = [("X_0 X_1", 1.0), ("Y_0 Y_1", 1.0), ("Z_0 Z_1", 1.0)]
+
+            hamiltonian = terms
+
+            self.assertSetEqual(set(hamiltonian), set(heisenberg_model.second_q_ops().to_list()))
+
+    def test_triangular(self):
+        """Test triangular lattice."""
+        triangle_graph = PyGraph(multigraph=False)
```

Comment:
Is there any reason not to use `qiskit_nature.second_q.properties.lattices.TriangularLattice`?

Reply:
I tried to use the `TriangularLattice`, but it seems that doesn't allow me to create just one triangle, it only create a pattern with triangles. If there is a way to create a graph with just one triangle using this class, please tell me how to do it.

Reply:
Hm yes you appear to be right about that. I was not aware of this limitation. In that case this is good as it is now :+1: